ci/deps.sh

```diff
@@ -17,48 +17,51 @@
 #    ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 #    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-#Do not delete
-set -e
+# Do not delete
+set -euo pipefail
 
-#wget Retries
+# wget Retries
 WGET_N_RETRIES=30
 WGET_WAIT_RETRIES_S=10
 
-#Don't pollute /
+# Don't pollute /
 mkdir -p /tmp
 cd /tmp
 
-#Dependencies (not fulfilled by Dockerfile)
+# Most of the software installation will probably honor this, which will improve perf
+export MAKEFLAGS="-j$(nproc)"
```

Comment:
* Although `$()` is POSIX, but since we still want to be `sh` compatible to avoid any _possible_ portability issues - JIC - please use ``
* It's rather impolite from us to take _all_ available lcores on the system for compilation, and really unnecessary for almost all modern CPUs. Besides, it can have unintended consequences if some parts would actually use _all_ of those. I am fine on adding parallelism, but a) should have reasonable defaults (e.g. 4 or whatever) b) having a way to override via env variable.

Reply:
>have reasonable defaults (e.g. 4 or whatever) 

On this, I was trying to research on what is the optimal `value` in `make -j <value>` in Github Actions. I know for Travis it was in the range of 2-4, but actually results were all similar, so we were using 2. We should probably start with 2 also.

Reply:
As commented in parallel, we will change this and we will honor a `NUM_WORKERS` env var which default will be declared in the `Dockerfile`

Reply:
This has been rewritten in the parallel PR. Resolving